codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/DocumentationConverter.java

```diff
@@ -133,18 +169,56 @@ private void writeText(TextNode node) {
 
             // Docs can have valid $ characters that shouldn't run through formatters.
             String text = node.text().replace("$", "$$");
-
             if (shouldStripPrefixWhitespace) {
                 shouldStripPrefixWhitespace = false;
                 text = StringUtils.stripStart(text, " \t");
             }
 
-            if (needsListPrefix) {
-                needsListPrefix = false;
-                writer.write("");
-                writeIndent();
-                text = "* " + StringUtils.stripStart(text, " \t");
+            String curString = writer.toString();
+            if (listDepth > 0) {
+                if (needsListPrefix) {
+                    needsListPrefix = false;
+                    StringBuilder sb = new StringBuilder();
+                    for (int i = 0; i < listDepth; i++) {
+                        sb.append("    ");
+                    }
+                    text = "- " + StringUtils.stripStart(text, " \t");
+                    text = sb.append(text).toString();
+                    writeNewline();
+                } else if (curString.charAt(curString.length() - 1) != ' ') {
```

Comment:
Consider adopting a discrete mechanism to avoid the extra branching here. It looks like we might be able to apply this to the main wrapping code below as well.

e.g. (rough code, not sure this is exactly what you want, but the intent should be clear)
```java
String joinWithEnsuredSeparator(String i, String j, String separator) {
    if (i.endsWith(separator) || j.startsWith(separator)) {
        return i + j;
    }
    return i + separator + j;
}
```

[Kotlin does something like this in their docs handler to ensure line / section breaks.](https://github.com/awslabs/smithy-kotlin/blob/main/codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/lang/DocumentationPreprocessor.kt#:~:text=private%20fun%20StringBuilder.ensureLineBreak()%20%7B)
(anchor link may not work, scroll to the bottom)